OpenOversight/app/commands.py

```diff
@@ -252,9 +252,12 @@ def process_assignment(row, officer, compare=False):
                 for fieldname in assignment_fieldnames:
                     current = getattr(assignment, fieldname)
                     # Test if fields match between row and existing assignment
-                    if (current and fieldname in row and row[fieldname] == current) or \
-                            (not current and (fieldname not in row or not row[fieldname])):
-                        i += 1
+                    try:
+                        if (current and fieldname in row and str(row[fieldname]) == str(current)) or \
+                                (not current and (fieldname not in row or not row[fieldname])):
+                            i += 1
+                    except ValueError as e:
+                        print('Problem comparing key via string casting: {} \n {}'.format(fieldname, e))
```

Comment:
OKAY. This was why tests were failing. Took a long while to suss out. So, effectively, we had `datetime.datetime(2020, 01, 01) == str('2020-01-01')` which isn't true, so a new row was being added. I stringify the value so that we have a more equal comparison, and catch and skip if we blow up (causing a spurious update, which I imagine is about the same place we are at as with the prior `==`)